command/agent/config_parse_test.go

```diff
@@ -196,15 +196,19 @@ var basicConfig = &Config{
 		},
 	},
 	Telemetry: &Telemetry{
```

Comment:
Judging from https://github.com/hashicorp/nomad/pull/20165 it's unfortunately easy to break config parsing with our default values. Should we add a test case that covers the "non-nil `Telemetry` block but with `in_memory_collection_interval` and `in_memory_retention_period` unset?

Reply:
Added something in: https://github.com/hashicorp/nomad/pull/20166/commits/a94d996cd8542922b1a9d0636e41d52f5da078d4

Is this what you had in mind?

Reply:
Perfect!